src/template.rs

```diff
@@ -15,6 +19,71 @@ pub struct FunctionTemplate<'a>(&'a isolate::Isolate, v8::FunctionTemplateRef);
 #[derive(Debug)]
 pub struct ObjectTemplate<'a>(&'a isolate::Isolate, v8::ObjectTemplateRef);
 
+#[derive(Debug)]
+pub struct Signature<'a>(&'a isolate::Isolate, v8::SignatureRef);
+
+/// A Signature specifies which receiver is valid for a function.
+impl<'a> Signature<'a> {
+    pub fn new(isolate: &'a isolate::Isolate) -> Signature<'a> {
+        let raw = unsafe {
+            util::invoke(isolate,
+                         |c| v8::Signature_New(c, isolate.as_raw(), ptr::null_mut()))
+                .unwrap()
+        };
+        Signature(isolate, raw)
+    }
+
+    pub fn new_with_receiver(isolate: &'a isolate::Isolate, receiver: &FunctionTemplate) -> Signature<'a> {
+        let raw = unsafe {
+            util::invoke(isolate,
+                         |c| v8::Signature_New(c, isolate.as_raw(), receiver.1))
+                .unwrap()
+        };
+        Signature(isolate, raw)
+    }
+}
+
+/// A FunctionTemplate is used to create functions at runtime. 
+/// There can only be one function created from a FunctionTemplate in a context.
+///
+/// Any modification of a FunctionTemplate after first instantiation will trigger a crash.
```

Comment:
I've changed get_function to take ownership. Initially, I was concerned that dropping the FunctionTemplate while the returned Function still lives on would cause an issue. However, it appears that the internal state created by FunctionTemplate is stored in the context and the returned Function references that.